platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/SignCommand.java

```diff
@@ -0,0 +1,156 @@
+     * directories
+     * <p>
+     * If an element of this list is a directory, then all files in that directory will be signed, recursively
+     * <p>
+     * Defaults to the current working directory
+     */
+    private List<Path> pathsToSign = List.of(FileUtils.getAbsolutePath());
+
+    /**
+     * The directory where signature files will be generated. Defaults to in-place signatures (null)
+     */
+    @Nullable
+    private Path destinationDirectory = null;
+
+    /**
+     * The path to the key file to use to generate signatures
+     */
+    private Path keyFilePath;
+
+    /**
+     * The password to the key file
+     */
+    private String keyFilePassword;
+
+    /**
+     * The alias of the key in the key file
+     */
+    private String keyAlias;
+
+    /**
+     * The key pair to use to generate signatures
+     */
+    private KeyPair keyPair;
+
+    @CommandLine.Parameters(description = "The path to the key file to use to generate signatures", index = "0")
+    private void setKeyFile(@NonNull final Path keyFilePath) {
+        pathMustExist(keyFilePath.toAbsolutePath());
+
+        this.keyFilePath = keyFilePath;
+    }
+
+    @CommandLine.Parameters(description = "The password to the key file", index = "1")
+    private void setKeyFilePassword(@NonNull final String keyFilePassword) {
+        this.keyFilePassword = keyFilePassword;
+    }
+
+    @CommandLine.Parameters(description = "The alias of the key in the key file", index = "2")
+    private void setKeyAlias(@NonNull final String keyAlias) {
+        this.keyAlias = keyAlias;
+    }
+
+    @CommandLine.Option(
+            names = {"-p", "--paths-to-sign"},
+            description = "The paths to what will be signed. Can contain single files, as well as directories."
+                    + "Defaults to the current working directory")
+    private void setPathsToSign(@NonNull final List<Path> pathsToSign) {
+        for (final Path path : pathsToSign) {
+            pathMustExist(path.toAbsolutePath());
+        }
+
+        this.pathsToSign = pathsToSign;
+    }
+
+    @CommandLine.Option(
+            names = {"-d", "--destination-directory"},
+            description = "Specify the destination directory where signature files will be generated."
+                    + "If not specified, a signature file will be generated in the same directory as the source file")
+    private void setDestinationDirectory(@NonNull final Path destinationDirectory) {
+        this.destinationDirectory = destinationDirectory.toAbsolutePath();
+    }
+
+    @Override
+    public Integer call() {
+        keyPair = FileSigningUtils.loadPfxKey(keyFilePath, keyFilePassword, keyAlias);
+
+        for (final Path path : pathsToSign) {
+            if (Files.isDirectory(path)) {
+                signAllFilesInDirectory(path);
+            } else {
+                generateSignatureFile(destinationDirectory, path, keyPair);
+            }
+        }
```

Comment:
Instead of iterating here, could we just call `signAllFilesInDirectory()` on each path to sign? If you don't like that we pass non-directories to this method, perhaps rename it to something like `signAllFilesAtPath()`.

Reply:
Yes, changed